From 3c3b7d2f7922edb584c89aae2f564d66a8037c8d Mon Sep 17 00:00:00 2001
From: Dan Williams <dcbw@redhat.com>
Date: Tue, 18 Jan 2011 18:29:56 -0600
Subject: [PATCH 103/122] wifi: clear wireless secrets tries counter when deactivating

If NM asks for secrets, and then a client calls ActivateDevice on
that same connection, the secrets tries counter doesn't get reset
and NM then thinks we need completely new secrets when we really
don't since the old secrets request isn't valid anymore.

Found by Evan Broder
---
 src/nm-device-wifi.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/src/nm-device-wifi.c b/src/nm-device-wifi.c
index b09d112..451284d 100644
--- a/src/nm-device-wifi.c
+++ b/src/nm-device-wifi.c
@@ -1231,6 +1231,15 @@ real_deactivate_quickly (NMDevice *dev)
 	NMDeviceWifi *self = NM_DEVICE_WIFI (dev);
 	NMDeviceWifiPrivate *priv = NM_DEVICE_WIFI_GET_PRIVATE (self);
 	NMAccessPoint *orig_ap = nm_device_wifi_get_activation_ap (self);
+	NMActRequest *req;
+	NMConnection *connection;
+
+	req = nm_device_get_act_request (dev);
+	if (req) {
+		connection = nm_act_request_get_connection (req);
+		/* Clear wireless secrets tries when deactivating */
+		g_object_set_data (G_OBJECT (connection), WIRELESS_SECRETS_TRIES, NULL);
+	}
 
 	cleanup_association_attempt (self, TRUE);
 
-- 
1.7.3.4

